lib/input/sketch/entities/layers/abstract_layer.dart

```diff
@@ -1,21 +1,24 @@
 import 'package:json_annotation/json_annotation.dart';
-import 'package:parabeac_core/input/entities/abstract_sketch_node_factory.dart';
-import 'package:parabeac_core/input/entities/objects/frame.dart';
-import 'package:parabeac_core/input/entities/style/style.dart';
+import 'package:parabeac_core/design_logic/design_node.dart';
+import 'package:parabeac_core/input/sketch/entities/abstract_sketch_node_factory.dart';
+import 'package:parabeac_core/input/sketch/entities/objects/frame.dart';
+import 'package:parabeac_core/input/sketch/entities/style/style.dart';
 import 'package:parabeac_core/interpret_and_optimize/entities/subclasses/pb_intermediate_node.dart';
 import 'package:parabeac_core/interpret_and_optimize/helpers/pb_context.dart';
 
 @JsonSerializable(nullable: false)
 // title: Abstract Layer
 // description: Abstract base schema for all layers
 // type: object
-abstract class SketchNode {
+abstract class SketchNode extends DesignNode {
```

Comment:
I think this should be an implementation not an extends. 🤔

Reply:
Good catch